libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp

```diff
@@ -235,6 +235,28 @@ int WiFiClientSecure::connect(const String& host, uint16_t port) {
   return connect(host.c_str(), port);
 }
 
+int WiFiClientSecure::connectAsync(const char* name, uint16_t port, bool* waiting) {
+  if (waiting != nullptr) *waiting = false;
+  if (connected()) return 1;
+
+  IPAddress remote_addr;
+
+  if (!WiFi.hostByNameAsync(name, remote_addr, waiting)) {
+    if (waiting != nullptr and *waiting == true) {
```

Comment:
For consistency, `and` -> `&&`, because it's that way everywhere else in the core.